source/ref_deployments.rst

```diff
@@ -0,0 +1,324 @@
+
+.. code-block:: bash
+
+    odahuflowctl deployment get --format json
+
+Getting the model name of the model deployments:
+
+.. code-block:: bash
+
+    odahuflowctl deployment get --id tensorflow-cli -o 'jsonpath=[*].spec.model.name'
+
+* Creating of a training from `train.yaml` file:
+
+.. code-block:: bash
+
+    odahuflowctl deployment create -f train.yaml
+
+* All model deployments commands and documentation:
+
+.. code-block:: bash
+
+    odahuflowctl deployment --help
+
+* All model deployments commands and documentation:
+
+.. code-block:: bash
+
+    odahuflowctl deployment --help
+
+* Getting a model deployment information:
+
+.. code-block:: bash
+
+    odahuflowctl model info --md wine
+
+* Making a prediction:
+
+.. code-block:: bash
+
+    odahuflowctl model invoke --md wine --file request.json
+
+JupyterLab
+----------
+
+Odahu-flow provides the :ref:`int_jupyterlab_extension:JupyterLab extension` for interacting with Model Deployments API.
+
+********************************************
+Service Catalog
+********************************************
+
+Service catalog provides a Swagger UI for Model Deployments.
+
+.. note::
+
+    The model must provide :ref:`input and output samples<ref_trainings:MLFlow protocol>` to appear in the Service Catalog
+
+Service catalog Swagger UI:
+
+.. image:: img/service_catalog.png
+
+Example of a prediction request:
+
+.. image:: img/service_catalog_predict.png
+
+********************************************
+Grafana Dashboard
+********************************************
+
+Out of the box, Odahu-flow provides the Grafana Model Deploymet dashboard.
+It contains the charts with following system metrics:
+
+    * availability
+    * replicas
+    * CPU
+    * memory
+    * number of failed HTTP requests
+    * latency
+    * ...
+
+Example of the dashboard:
+
+.. image:: img/model_deployment_dashboard.png
+
+********************************************
+Feedback
+********************************************
+
+**Model Feedback** provides a view of performance over all stages of model lifecycle.
+
+The mechanism is simple:
+
+1. Ask Deploy for prediction (with or without ``Request-Id`` provided)
+2. Send Feedback to Odahu-flow about the prediction (with ``Request-Id`` returned from previous step)
+3. Odahu-flow stores the prediction and feedback to a configurable location
+
+.. important::
+
+   This flow requires ``feedback`` to be enabled in ``values.yaml`` during Helm chart install
+
+Protocol
+--------
+
+1. If prediction is requested without Request-ID: ``Request-ID`` header with random ID is added to the request. Otherwise, Request-ID is not generated.
+2. Request and response are stored on configured external storage (eg. S3, GCS)
+3. User sends Model Feedback as an argument to the feedback endpoint. (Feedback can be arbitrary JSON.)
+5. All Feedback is persisted on external storage and can be used by models during subsequent Trains.
+
+Worked example
```

Comment:
Should it be `working example` or `work example` ?

Reply:
Fixed